### PR TITLE
Update footer and navbar to match PennyLane website

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**

The PennyLane website is updating the styling of its footer and navbar. For a consistent experience, it makes sense to also update the layout, styling, etc. of the PennyLane Docs pages to match the main website.

**Description of the Change:**

Updated the inherited Xanadu Sphinx Theme (XST) version to v0.4.0 as well as the default navbar and footer options. The technical details of these changes (e.g., CSS modifications) are captured in a [different PR to the XST repository](https://github.com/XanaduAI/xanadu-sphinx-theme/pull/40).

<details><summary>Desktop Navbar</summary>
<p>

Desktop Navbar

</p>
</details> 

<details><summary>Desktop Footer</summary>
<p>

Desktop Footer

</p>
</details> 

<details><summary>Mobile Navbar</summary>
<p>

Mobile Navbar Expanded

Mobile Navbar ToC Expanded

</p>
</details> 

<details><summary>Mobile Footer</summary>
<p>

Mobile Footer

</p>
</details> 

**Benefits:**

* The navbar and footer on the PennyLane Docs pages will be consistent with the main PennyLane website.

**Possible Drawbacks:**

* Some accessibility features were compromised in the XST v0.4.0 release; however, these will be addressed in a future patch release.

**Related GitHub Issues:**

None.

--- 

**Preview:** https://xanaduai-pennylane--29.com.readthedocs.build/projects/sphinx-theme/en/29/.

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
-## Release 0.4.0 (development release)
+## Release 0.4.0 (current release)
+
+### Improvements
+
+* Updated the navbar and footer configuration and styles to match the PennyLane website.
+  [(#XXX)](https://github.com/PennyLaneAI/pennylane-sphinx-theme/pull/XXX)
```
Comment (Mandrenkov) [inline .github/CHANGELOG.md:6]:
```suggestion
  [(#29)](https://github.com/PennyLaneAI/pennylane-sphinx-theme/pull/29)
```

**Review by Lorraine-at-Xanadu — CHANGES_REQUESTED** [review]
Hey @Mandrenkov! You've done an incredible job matching the navbar in Sphinx. 

1. Learn > Glossary: Link is broken and should go to https://pennylane.ai/qml/glossary
2. When I hover on 'Learn' (for example) and then hover on 'Documentation' there's a delay when the 'Learn' menu closes causing overlap. 

image

**Review by Lorraine-at-Xanadu — CHANGES_REQUESTED** [review]
The footer looks good, too, @Mandrenkov. A couple of things...

1. 'FAQ' is missing after 'Glossary' under _For learners_ (https://pennylane.ai/faq).
2. The 'Codebook' link doesn't open to https://codebook.xanadu.ai/ under _For learners_.

**Comment by Mandrenkov** [comment]
Thanks for the review, @Lorraine-at-Xanadu! I've implemented all the feedback (great catches by the way!).

**Review by josh146 — APPROVED** [review]
😍

**Review by Lorraine-at-Xanadu — APPROVED** [review]
This looks great! Thank you, @Mandrenkov for making the navbar on the Docs pages match the rest of the site.

**Review by ashishks0522 — COMMENTED** [review]
This is looking great @Mandrenkov!

One different behaviour I noticed is that on [PenyLane NavBar ](https://dev.pennylane.ai/pull_request_previews/605/) we open external links except to docs pages (like to github, discussion fourm etc) in new tab and here we are opening everything in same tab.

I guess this could be more of a product question as well, should we change that here or in the react one?

cc: @Lorraine-at-Xanadu

**Comment by Lorraine-at-Xanadu** [comment]
> This is looking great @Mandrenkov!
> 
> One different behaviour I noticed is that on [PenyLane NavBar ](https://dev.pennylane.ai/pull_request_previews/605/) we open external links except to docs pages (like to github, discussion fourm etc) in new tab and here we are opening everything in same tab.
> 
> I guess this could be more of a product question as well, should we change that here or in the react one?
> 
> cc: @Lorraine-at-Xanadu

Hi @ashishks0522 and @Mandrenkov! Based on @WillXanadu's design, only the Codebook and Support open in a new tab because those are the only places I see this symbol:
image

**Comment by Mandrenkov** [comment]
Nice catch, @ashishks0522! I've updated the external links to open in a new tab (cc @Lorraine-at-Xanadu).

**Comment by WillXanadu** [comment]
this looks great @Mandrenkov ! A slight change that was added recently was to change the hover state of the root nav links to blue, instead of the current light grey. Is that something you can easily add in this PR?

**Comment by ashishks0522** [comment]
Probably the last change, but as mentioned [here](https://github.com/XanaduAI/pennylane.ai-react/pull/621#pullrequestreview-1638536727) - we would have to remove the `Get Involved` link from the footer 😅

**Comment by Mandrenkov** [comment]
> Probably the last change, but as mentioned [here](https://github.com/XanaduAI/pennylane.ai-react/pull/621#pullrequestreview-1638536727) - we would have to remove the `Get Involved` link from the footer 😅

Thanks, @ashishks0522! This has been implemented [here](https://github.com/PennyLaneAI/pennylane-sphinx-theme/pull/29/commits/bf6fc292facdc3d5b3989b99f7a849a8f196541a).

**Review by ashishks0522 — APPROVED** [review]

**Review by Lorraine-at-Xanadu — APPROVED** [review]
Hey @Mandrenkov! I double-checked that the changes on the REACT navbar and footer are also working here. This looks great! Thank you.

**Pull request merged**